nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -589,11 +577,14 @@ define(
                 }
                 model.setItem('user-settings.batchMode', newBatchMode);
                 bus.emit('set-batch-mode', newBatchMode);
-                toggleTab('configure').then(() => {
-                    toggleTab('configure');
-                });
                 evaluateAppState();
-                // if the configure widget is selected, stop and start it.
+
+                if (tabSet.selectedTab && tabSet.selectedTab.id === 'configure') {
+                    // if the configure widget is selected, stop and start it.
+                    toggleTab('configure').then(() => {
+                        toggleTab('configure');
+                    });
+                }
```

Comment:
Changed this code so that it does what the comment suggests. As it was, you can switch to batch mode when you have the Info tab selected, and all hell breaks loose.